server/catgenome/src/main/java/com/epam/catgenome/manager/blast/BlastTaskManager.java

```
@@ -93,28 +96,48 @@ private void loadTaskParameters(final BlastTask blastTask) {
+            && (taskVO.getExcludedOrganisms() == null || taskVO.getExcludedOrganisms().isEmpty())) {
+            Assert.isTrue(false, MessageHelper.getMessage(MessagesConstants.ERROR_BLAST_ORGANISMS));
+        }
         BlastRequest blastRequest = new BlastRequest();
-        blastRequest.setAlgorithm(blastTask.getAlgorithm());
-        blastRequest.setBlastTool(blastTask.getExecutable());
-        blastRequest.setDbName(blastTask.getDatabase());
-        blastRequest.setQuery(blastTask.getQuery());
-        blastRequest.setOptions(blastTask.getOptions());
-        blastRequest.setTaxIds(blastTask.getOrganisms());
-        blastRequest.setExcludedTaxIds(blastTask.getExcludedOrganisms());
-        blastRequest.setMaxTargetSequence(MAX_TARGET_SEQUENCE);
-        blastRequest.setExpectedThreshold(EXPECTED_THRESHOLD);
+        blastRequest.setAlgorithm(taskVO.getAlgorithm());
+        blastRequest.setBlastTool(taskVO.getExecutable());
+        blastRequest.setDbName(taskVO.getDatabase());
+        blastRequest.setQuery(taskVO.getQuery());
+        blastRequest.setOptions(taskVO.getOptions());
+        blastRequest.setTaxIds(taskVO.getOrganisms());
+        blastRequest.setExcludedTaxIds(taskVO.getExcludedOrganisms());
+        Assert.isTrue(taskVO.getParameters().containsKey(MAX_TARGET_SEQUENCE),
+                MessageHelper.getMessage(MessagesConstants.ERROR_BLAST_TASK_PARAMETERS));
+        Assert.isTrue(taskVO.getParameters().containsKey(EXPECTED_THRESHOLD),
+                MessageHelper.getMessage(MessagesConstants.ERROR_BLAST_TASK_PARAMETERS));
+        blastRequest.setMaxTargetSequence(Long.parseLong(taskVO.getParameters().get(MAX_TARGET_SEQUENCE)));
+        blastRequest.setExpectedThreshold(Long.parseLong(taskVO.getParameters().get(EXPECTED_THRESHOLD)));
         BlastRequestInfo blastRequestInfo = blastRequestManager.createTask(blastRequest);
         if (!blastRequestInfo.getStatus().equals("ERROR")) {
```

Let's throw an error if `blast-wrapper` returned an error, since we do not persist task in this case